### PR TITLE
add mpi support for emcee and kombine

### PR DESCRIPTION
**Pull request by ahnitz**
This adds the ability to run with kombine or emcee on multiple nodes. 

You submit a job as 

mpirun -np ${NUM_PROC+1} pycbc_inference .......   --num-processes $NUM_PROC --use-mpi

To run on multiple nodes you can use the mpirun --hostfile option, or --host where you list the machines on the command line. We should investigate the parallel universe as well for launching on a large number of cores through condor.

### CODE REVIEW

**Comment by ahnitz** [comment]
Rebased from the latest master. This depends on a patch to Kombine however as well, although emcee will work as is.

**Comment by ahnitz** [comment]
I don't think there is anything I can do about the codacy warning. Unfortunately, we do need to rely on a globally accessible variable here to coordinate.

**Comment by ahnitz** [comment]
I've rebased, and this still seems to run at least. 

There's a performance bottleneck, that is resolved in an extreme package pull request. See https://github.com/adrn/schwimmbad/pull/8

**Comment by ahnitz** [comment]
Ok, I've rebased from master and tested the following.

1) Code executes on current version of kombine
2) Code executes on improved version of kombine with reduced memory transfers
3) Code executes under mpi with both kombine and emcee

**Comment by ahnitz** [comment]
@cdcapano @cmbiwer Please have a look now if this can be merged.

**Comment by cdcapano** [comment]
@ahnitz Any idea why the travis build failed?

**Comment by ahnitz** [comment]
@cdcapano There, it passes now.

File: pycbc/inference/sampler_emcee.py [hunk 1]
```diff
@@ -52,40 +51,36 @@ class EmceeEnsembleSampler(BaseMCMCSampler):
     nwalkers : int
         Number of walkers to use in sampler.
     processes : {None, int}
-        Number of processes to use with multiprocessing. If None, all available
+        Numbfer of processes to use with multiprocessing. If None, all available
```
Comment (cdcapano) [inline pycbc/inference/sampler_emcee.py]: Looks like you have a typo here... unless "Numbfer" is a technical term.

**Comment by spxiwh** [comment]
Landscape is making a number of complaints (*some* of which look genuine) about this pull request. Can @ahnitz or @cdcapano look at this and comment here for prosperity:

https://landscape.io/github/ligo-cbc/pycbc/1449/diff

**Comment by ahnitz** [comment]
@spxiwh Ian, you are correct! There is a problem here. All the 'not callable' stuff isn't a problem, as we require the use of some global variables which have an unkown ability to be called when landscape reads the code. There unfortunately, isn't a simpler way to do this that we've thought of yet. *However* the 'processes' undefined variable is a bug! Somehow, I didn't think landscape had completed. Did it just catch up recently? Either that or I am blind. Thank you for reviewing this!

**Pull request merged**